src/Consumer.cc

```diff
@@ -82,9 +96,15 @@ void MessageListener(pulsar_consumer_t *rawConsumer, pulsar_message_t *rawMessag
     return;
   }
 
-  MessageListenerProxyData *dataPtr = new MessageListenerProxyData(cMessage, consumer);
+  std::promise<void> promise;
+  std::future<void> future = promise.get_future();
+  MessageListenerProxyData *dataPtr =
+      new MessageListenerProxyData(cMessage, consumer, [&]() { promise.set_value(); });
   listenerCallback->callback.BlockingCall(dataPtr, MessageListenerProxy);
   listenerCallback->callback.Release();
+
+  future.wait();
+  delete dataPtr;
```

Comment:
It's better not to use `delete` directly, use `std::unique_ptr`.

Reply:
The `BlockingCall` only accepts the raw pointer. FYI: https://github.com/nodejs/node-addon-api/blob/main/doc/threadsafe_function.md#blockingcall--nonblockingcall

Reply:
I mean use a RAII coding style. For example, if we have a function that only accepts a raw pointer:

```c++
void f(int*);
```

If now we have a raw pointer `p`, we can write:

```c++
std::unique_ptr<int> up{p};  // don't do anything with up
f(p);
// After going out of the scope, `delete p` will be called automatically.
```

If now we have a unique pointer `up`, we can write:

```c++
f(up.get());
// After going out of the scope, `delete up.get()` will be called automatically.
```